pnpm install with version 8 throws "unsupported engine" error

My current pnpm workspace uses pnpm version 8. Didn't migrate yet because it's a breaking change - v9 uses a different lock file.

When running `pnpm install trpc-rtk-query --engine-strict` it throws the following error:

```
 ERR_PNPM_UNSUPPORTED_ENGINE  Unsupported environment (bad pnpm and/or Node.js version)

Your pnpm version is incompatible with "/trpc-rtk-query/0.1.1(@reduxjs/toolkit@2.2.1)(@trpc/client@10.45.2)(@trpc/server@10.45.2)".

Expected version: >=9
Got: 8.10.5

This is happening because the package's manifest has an engines.pnpm field specified.
To fix this issue, install the required pnpm version globally.

To install the latest version of pnpm, run "pnpm i -g pnpm".
To check your pnpm version, run "pnpm -v".
```

It should be completely fine to install `trpc-rtk-query` with other versions of pnpm.

My suggestion: remove the `pnpm` from the `engines` field, and set the `packageManager` field to use [corepack](https://nodejs.org/docs/latest-v20.x/api/corepack.html).

--- a/.changeset/afraid-bats-double.md
+++ b/.changeset/afraid-bats-double.md
@@ -1,0 +1,5 @@
+---
+"trpc-rtk-query": minor
+---
+
+Set packageManager field instead of engines.pnpm

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -13,8 +13,6 @@ jobs:
 
       - name: Setup pnpm
         uses: pnpm/action-setup@v4
-        with:
-          version: 9.4.0
 
       - name: Setup node
         uses: actions/setup-node@v4

--- a/.github/workflows/release.yml
+++ b/.github/workflows/release.yml
@@ -22,8 +22,6 @@ jobs:
 
       - name: Setup pnpm
         uses: pnpm/action-setup@v4
-        with:
-          version: 9.4.0
 
       - name: Setup node
         uses: actions/setup-node@v4

--- a/package.json
+++ b/package.json
@@ -74,7 +74,7 @@
     "is-what": "^4.1.16"
   },
   "engines": {
-    "node": ">=20",
-    "pnpm": ">=9"
-  }
+    "node": ">=20"
+  },
+  "packageManager": "pnpm@9.6.0"
 }

